### PR TITLE
Expand motivation for `buildLfPackageGraph`

### PR DESCRIPTION
**Pull request by akrmn**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs [hunk 1]
```diff
@@ -540,8 +540,23 @@ data BuildLfPackageGraphArgs = BuildLfPackageGraphArgs
   }
 
 -- | The graph will have an edge from package A to package B:
---   - if A depends on B _OR_
---   - if A is a data-dependency, B is a (regular) dependency, and B doesn't depend on any data-dependencies.
+--   1. if A depends on B _OR_
+--   2. if A is a data-dependency, B is a (regular) dependency, and B doesn't
+--      depend (directly or transitively) on any data-dependencies.
+--
+--   The reason we want the second kind of edge is to ensure that the instances
```
Comment (sofiafaro-da) [inline compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs:547]: It's not just instances that get rewritten. Typeclass definitions get turned into re-exports of compatible typeclasses from the dependencies (`Eq` from a data-dependency becomes `Eq` in the current stdlib). Not sure how to explain that here / how much detail to get into.

**Review by sofiafaro-da — APPROVED** [review]
Thank you!

File: compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs (hunk 1 above)
Comment (akrmn) [reply compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs]: thanks! I reworded a bit, now it's "uses of type classes" that get rewritten, with a parenthetical explaining that includes (type class) definitions, instances, constraints and exports.

Comment (sofiafaro-da) [reply compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs]: Nice, thank you!

**Review by robin-da — APPROVED** [review]
Thanks a lot!

**Pull request merged**